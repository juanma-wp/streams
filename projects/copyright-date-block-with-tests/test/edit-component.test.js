import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

const mockEditProps = {
  attributes: {
    fallbackCurrentYear: "2024",
    showStartingYear: false,
    startingYear: "",
  },
  setAttributes: jest.fn(),
  clientId: "test-id",
  className: "wp-block-copyright-date",
};

const originalDate = global.Date;

import Edit from "../src/edit";

describe("Edit component", () => {
  beforeEach(() => {
    mockEditProps.setAttributes.mockClear();
    global.Date = class extends Date {
      getFullYear() {
        return 2024;
      }
    };
  });
  afterEach(() => {
    global.Date = originalDate;
  });

  it("matches snapshot", () => {
    const { container } = render(<Edit {...mockEditProps} />);
    expect(container).toMatchSnapshot();
  });
  it("displays current year by default", () => {
    const { getByText } = render(<Edit {...mockEditProps} />);
    expect(getByText(/Â© 2024/)).toBeInTheDocument();
  });
  describe("Settings Panel", () => {
    it("renders Inspector Controls", () => {
      const { getByTestId } = render(<Edit {...mockEditProps} />);
      expect(getByTestId("inspector-controls")).toBeInTheDocument();
    });
    it("renders PanelBody with correct title", () => {
      const { getByTestId } = render(<Edit {...mockEditProps} />);
      expect(getByTestId("panel-body")).toBeInTheDocument();
      expect(getByTestId("panel-body")).toHaveAttribute(
        "data-title",
        "Settings"
      );
    });
    it("doesn't render TextControl if showStartingYear is false", () => {
      const { queryByTestId } = render(<Edit {...mockEditProps} />);
      expect(queryByTestId("text-control")).not.toBeInTheDocument();
    });
    it("renders TextControl if showStartingYear is true", () => {
      const mockEditPropsWithStartingYear = {
        ...mockEditProps,
        attributes: {
          ...mockEditProps.attributes,
          showStartingYear: true,
        },
      };
      const { getByTestId } = render(
        <Edit {...mockEditPropsWithStartingYear} />
      );
      expect(getByTestId("text-control")).toBeInTheDocument();
    });
  });
});
