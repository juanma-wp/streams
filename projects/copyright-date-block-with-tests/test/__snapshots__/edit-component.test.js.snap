// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit component matches snapshot 1`] = `
<div>
  <div
    data-testid="inspector-controls"
  >
    <div
      data-testid="panel-body"
      data-title="Settings"
    >
      <div
        data-testid="toggle-control"
      />
    </div>
  </div>
  <p
    class="wp-block-copyright-date"
  >
    Â© 
    2024
  </p>
</div>
`;
